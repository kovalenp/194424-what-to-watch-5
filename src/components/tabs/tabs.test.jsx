import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import Tabs from "./tabs";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

describe(`Tabs meets B21`, () => {

  it(`and rendered correctly for auth'ed`, () => {
    const tree = renderer.create((
      <Tabs
        active={``}
        onActiveChange={() => { }}
      >
        <MockComponent />
        <MockComponent />
        <MockComponent />
      </Tabs>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
