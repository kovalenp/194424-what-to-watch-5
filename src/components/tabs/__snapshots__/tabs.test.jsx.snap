// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs meets B21 and rendered correctly for auth'ed 1`] = `
<nav
  className="movie-nav movie-card__nav"
>
  <ul
    className="movie-nav__list"
  >
    <li
      className="movie-nav__item"
    >
      <a
        className="movie-nav__link"
        href="#"
        onClick={[Function]}
      />
    </li>
    <li
      className="movie-nav__item"
    >
      <a
        className="movie-nav__link"
        href="#"
        onClick={[Function]}
      />
    </li>
    <li
      className="movie-nav__item"
    >
      <a
        className="movie-nav__link"
        href="#"
        onClick={[Function]}
      />
    </li>
  </ul>
</nav>
`;
